### ISSUE TITLE
Cannot extend native Error class

### ISSUE THREAD

**Issue by ianbytchek** [Microsoft/TypeScript#7639]
```ts
class FooError extends Error {
    constructor(message:string) {
        super(message);
    }
}
console.log((new FooError('foo')).message); // <nothing>
console.log((new Error('bar')).message); // bar
```

Am I missing something obvious or is there a bug with 1.8.9?

**Comment by mhegazy** [comment]
looks like a duplicate of https://github.com/Microsoft/TypeScript/issues/5069, please see the discussion in https://github.com/Microsoft/TypeScript/issues/1168#issuecomment-107833988

**Comment by ianbytchek** [comment]
I did. @ahejlsberg says:

> Fixed by #3516. Classes can now extend arbitrary expressions of constructor function types.

Relevant [PR](https://github.com/Microsoft/TypeScript/pull/3516) contains an example:

```ts
class MyError extends Error { }
```

**Comment by mhegazy** [comment]
it can. the compiler does not complain at you any more. the problem is not the TS compiler.

the problem here is that these classes are native, your runtime does not support extending them at the moment. In the future i would expect all these native types to be conformant with ES6 spec. @unional has a comment on how to get it to work with the engines today: https://github.com/Microsoft/TypeScript/issues/1168#issuecomment-107833988.

**Comment by unional** [comment]
@mhegazy , wow, you got good memory. :+1:

**Comment by ianbytchek** [comment]
Ok, I feel a little awkward here, but that example doesn't compile:

```
'super' must be called before accessing 'this' in the constructor of a derived class. (17009)
Type 'Error | string' is not assignable to type 'Error'.
```

From what I can take from there, the suggested solution is to simply do `this.message = message` in the constructor?

**Comment by unional** [comment]
The first line is a guard. You can take it out. The compiler has changed since then.
